src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java

```diff
@@ -168,6 +186,88 @@ private String getResponsePhrase(String name, boolean familiar) {
         }
     }
 
+    private String getRoboyFactsPhrase(Roboy roboy) {
+        String result = "";
+
+        // Get some random properties facts
+        if (roboy.getProperties() != null && !roboy.getProperties().isEmpty()) {
+            HashMap<String, Object> properties = roboy.getProperties();
+            if (properties.containsKey(full_name.type)) {
+                result += " " + String.format(infoValues.getSuccessAnswers(full_name).getRandomElement(), properties.get(full_name.type));
+            }
+            if (properties.containsKey(birthdate.type)) {
+                result += " " + String.format(infoValues.getSuccessAnswers(age).getRandomElement(), determineAge(properties.get(birthdate.type).toString()));
+            } else if (properties.containsKey(age.type)) {
+                result += " " + String.format(infoValues.getSuccessAnswers(age).getRandomElement(), properties.get(age.type) + " years!");
+            }
+            if (properties.containsKey(skills.type)) {
+                RandomList<String> retrievedResult = new RandomList<>(Arrays.asList(properties.get("skills").toString().split(",")));
+                result += " " + String.format(infoValues.getSuccessAnswers(skills).getRandomElement(), retrievedResult.getRandomElement());
+            }
+            if (properties.containsKey(abilities.type)) {
+                RandomList<String> retrievedResult = new RandomList<>(Arrays.asList(properties.get("abilities").toString().split(",")));
+                result += " " + String.format(infoValues.getSuccessAnswers(abilities).getRandomElement(), retrievedResult.getRandomElement());
+            }
+            if (properties.containsKey(future.type)) {
+                RandomList<String> retrievedResult = new RandomList<>(Arrays.asList(properties.get("future").toString().split(",")));
+                result += " " + String.format(infoValues.getSuccessAnswers(future).getRandomElement(), retrievedResult.getRandomElement());
+            }
+        }
+
+        if (result.equals("")) {
+            result = "I am Roboy 2.0! ";
+        }
+
+        // Get a random relationship fact
+        Neo4jRelationship predicate = roboyRelatioshipPredicates.getRandomElement();
+        MemoryNodeModel node = getMemNodesByIds(roboy.getRelationships(predicate)).getRandomElement();
+        if (node != null) {
+            String nodeName = "";
+            if (node.getProperties().containsKey("full_name") && !node.getProperties().get("full_name").equals("")) {
+                nodeName = node.getProperties().get("full_name").toString();
+            } else {
+                nodeName = node.getProperties().get("name").toString();
+            }
+            result += " " + String.format(infoValues.getSuccessAnswers(predicate).getRandomElement(), nodeName);
+        }
+
+        return result;
+    }
+
+    /**
+     * A helper function to determine the age based on the birthdate
+     *
+     * Java 8 specific
```

Comment:
what exactly is java 8 specific?

Reply:
Because of the used java.time.* functions. Moved to the util

Reply:
ah, you mean it requires java 8+
I think we use more java 8 stuff in some other places anyway
btw,  java 10 is planned to be released this september so we don't have to assume that java 8 is new